typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/defaultErrorHandlerDecoratorStrategy.ts

```diff
@@ -34,6 +35,47 @@ import { GraphNode } from "./helpers/graph";
  * ```
  */
 export default class DefaultErrorHandlerDecoratorStrategy implements CanvasItemDecoratorStrategy {
+	/** Adjustment to canvas [x,y] coordinates to compensate for item icon anchor position */
+	private static readonly OFFSET: [number, number] = [40, -10];
+
+	/**
+	 * Extracts the name of thedefault error handler in an Array to be added to the start node names of a Graph.
+	 * @param {WorkflowItemDescriptor[]} items - workflow items
+	 * @param {GraphNode[]} nodes - graph nodes mapped to the WF items, with name=item# and targets populated.
+	 * @returns {string[]} an array containing the node name (item#) of the default error handler. Empty if there is none.
+	 * @throws Error if there are more than 1 Default Error Handler items
+	 * @throws Error if the Default Error Handler item is a target of another node
+	 * @throws Error if the Default Error Handler has more than 1 target or the target node cannot be found
+	 * (e.g. the default "end" node)
+	 */
+	public static getDefaultErrorHandlerNodes(items: WorkflowItemDescriptor[], nodes: GraphNode[]): string[] {
```

Comment:
Initially the role of the "graph.ts" file was to hold all of the logic of calculating the position of the nodes. There were a few "hacky" instances where I had to add some dummy elements, but the way I saw that was it was part of the forming of the initial nodes array. One more hacky maybe moment was the part where I manually set the start node position.

I see a few issues breaking that/related to it:

1. formatPosition is something that needs to be part of the graph.ts. Or rather, when we get to the decorator, the decorator should accept them AS IS and do nothing more.

2. OFFSET and the likes are, again, not part of `graph.ts` is a bad idea, We could instead pass the offset somehow to the node, when we are creating the initial structure to pass to the `graph.ts`?

3. `origName` is not needed. I don't really understand sill why it's needed but it doesn't matter. The name of the node should be more than enough and more than unique. 

4. This function `getDefaultErrorHandlerNodes` - does validation we don't want to do. I mean here that we don't want to validate stuff outside of "orphaned" nodes. Even if we did, this kind of validation is not for this function. I can be convinced if you move the validation outside somewhere idk.

Reply:
1 - will remove offset from formatPosition, it will be calculated in Graph instead
2 - strategies will include offset in the nodes (getGraphNode) to be used in Graph^
3 - added for error-handling purposes. While item1, item2 ... are unique names, they aren't as informative as seeing which method has an error in its targets.
4 - Will move to a separate validator, accessible via the strategy.